lib/logmatcher.c

```diff
@@ -246,9 +261,9 @@ log_matcher_string_compile(LogMatcher *s, const gchar *pattern, GError **error)
   LogMatcherString *self = (LogMatcherString *) s;
 
   g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+  log_matcher_store_pattern(s, pattern);
 
-  self->pattern = g_strdup(pattern);
-  self->pattern_len = strlen(self->pattern);
+  self->pattern_len = strlen(pattern);
```

Comment:
This can't go to store_pattern as that is a base class which does not have a pattern_len member.

the pattern_len is only needed for the string matching and not regexp ones.